src/groupWith.js

```diff
@@ -0,0 +1,33 @@
+var _curry2 = require('./internal/_curry2');
+
+/**
+ * Takes a list and returns a list of lists where each sublist contains only equal elements.
```

Comment:
> contains only elements which satisfy

This is suggestive of *filtering*.

This would be clearer, I think:

_Takes a list and returns a list of lists where each sublist's elements are all "equal" according to the provided equality function._